core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -525,22 +519,40 @@ private[spark] class MemoryStore(blockManager: BlockManager, maxMemory: Long)
   /**
    * Reserve the unroll memory of current unroll successful block used by this task
    * until actually put the block into memory entry.
+   * @return whether the request is granted.
    */
-  def reservePendingUnrollMemoryForThisTask(memory: Long): Unit = {
+  private def reservePendingUnrollMemoryForThisTask(
+      blockId: BlockId,
+      memory: Long,
+      droppedBlocks: mutable.Buffer[(BlockId, BlockStatus)]): Boolean = {
     val taskAttemptId = currentTaskAttemptId()
     accountingLock.synchronized {
-       pendingUnrollMemoryMap(taskAttemptId) =
-         pendingUnrollMemoryMap.getOrElse(taskAttemptId, 0L) + memory
+      val acquired = memoryManager.acquireUnrollMemory(blockId, memory, droppedBlocks)
+      val success = acquired == memory
+      if (success) {
+        pendingUnrollMemoryMap(taskAttemptId) =
+          pendingUnrollMemoryMap.getOrElse(taskAttemptId, 0L) + memory
+      }
```

Comment:
I just noticed that the return value of `reservePendingUnrollMemoryForThisTask` is never checked. Does this mean that we never expect this operation to fail?